src/prommis/hydrogen_decrepitation/hydrogen_decrepitation_furnace.py

```diff
@@ -0,0 +1,1651 @@
+            self.flowsheet().config.time,
+            self.impurity_list,
+            doc="mass flow rate of individual impurity minerals [kg/s]",
+        )
+        def flow_mass_comp_impurity_feed(b, t, i):
+            return b.solid_in[t].flow_mass * b.solid_in[t].mass_frac_comp[i]
+
+        # mole flow rates of individual mineral impurity
+        @self.Expression(
+            self.flowsheet().config.time,
+            self.impurity_list,
+            doc="mole flow rate of individual impurity minerals [mol/s]",
+        )
+        def flow_mol_comp_impurity_feed(b, t, i):
+            return b.flow_mass_comp_impurity_feed[t, i] / b.mw_comp_impurity[i]
+
+        # the gas product is H2, no other species are carried out with it
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.config.gas_property_package.component_list,
+            doc="component flow of outlet gas stream",
+        )
+        def flow_mol_outlet_eqn(b, t, i):
+            return b.gas_out[t].flow_mol_comp[i] == b.gas_in[t].flow_mol_comp[i]
+
+        # total impurity solid product based on decrepitation reactions
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.product_list,
+            doc="mole flow rate of total product",
+        )
+        def flow_mol_comp_product_total_eqn(b, t, i):
+            if i == "Nd" or i == "Nd2Fe14B":
+                return (
+                    b.flow_mol_comp_product_total[t, i]
+                    == b.flow_mol_comp_impurity_feed[t, i]
+                )
+
+        # mole flow rate of recovered solid product stream
+        @self.Constraint(
+            self.flowsheet().config.time,
+            self.product_list,
+            doc="mole flow rate of recovered product stream",
+        )
+        def flow_mol_comp_product_recovered_eqn(b, t, i):
+            return (
+                b.flow_mol_comp_product_recovered[t, i]
+                == b.flow_mol_comp_product_total[t, i]
+            )
+
+        # mass flow rate of solid product recovered
+        @self.Constraint(self.flowsheet().config.time, doc="mass flow rate of product")
+        def flow_mass_product_recovered_eqn(b, t):
+            return b.flow_mass_product_recovered[t] == sum(
+                b.flow_mol_comp_product_recovered[t, i] * b.mw_comp_product[i]
+                for i in b.product_list
+            )
+
+        # mass fraction of impurity metal in the solid feed stream
+        @self.Expression(
+            self.flowsheet().config.time,
+            self.impurity_ele_list,
+            doc="mass fraction of impurity element in solid feed stream",
+        )
+        def mass_frac_comp_impurity_ele_feed(b, t, i):
+            if i == "Fe":
+                return (
+                    b.flow_mol_comp_impurity_feed[t, "Nd2Fe14B"]
+                    * b.am_Fe
+                    * 14
+                    / b.flow_mass_impurity_feed[t]
+                )
```

Comment:
I don't like the magic number 14 appearing in this expression. What does it represent?

Reply:
Why aren't we using a stoichiometry dictionary then?

Reply:
Probably should be. I'll add one.